src/components/AuthorViewpoint/ViewpointCard.tsx

```diff
@@ -0,0 +1,134 @@
+"use client";
+import { TrashIcon, PlusIcon } from "@heroicons/react/24/outline";
+import { Button, TextInput } from "@mantine/core";
+import { useState, useRef, useEffect } from "react";
+import Link from "next/link";
+
+type ViewpointCardProps = {
+    issueId: string;
+    viewpointTitle: string;
+    setViewpointTitle: (value: string) => void;
+    setViewpointContent: (value: string) => void;
+    publishViewpoint: () => void;
+};
+
+export default function ViewpointCard({
+    issueId,
+    viewpointTitle,
+    setViewpointTitle,
+    setViewpointContent,
+    publishViewpoint,
+}: ViewpointCardProps) {
+    const [contentEmpty, setContentEmpty] = useState<boolean>(true);
+    const inputRef = useRef<HTMLDivElement>(null);
+
+    useEffect(() => {
+        //manage the placeholder in the content area
+        if (inputRef?.current === null || inputRef.current.innerHTML !== "")
+            return;
+        const placeholderElement = document.createElement("p");
+        placeholderElement.className = "text-neutral-500";
+        placeholderElement.textContent =
+            "開始打字，或選取一段文字來新增引注資料";
+        inputRef.current.appendChild(placeholderElement);
+    }, [inputRef]);
+
+    const onPublish = () => {
+        if (viewpointTitle == "" || contentEmpty) {
+            console.log("Title and Content MUST NOT be empty");
+            return;
+        }
+
+        setViewpointContent(
+            inputRef.current?.textContent ? inputRef.current?.textContent : "",
+        );
+
+        publishViewpoint();
+    };
+
+    return (
+        <div className="flex h-full flex-col gap-2 overflow-auto rounded-lg bg-neutral-100 px-7 py-4">
+            <h1 className="text-lg font-semibold text-neutral-700">觀點</h1>
+            <TextInput
+                value={viewpointTitle}
+                onChange={(e) => setViewpointTitle(e.currentTarget.value)}
+                variant="unstyled"
+                radius={0}
+                placeholder="用一句話簡述你的觀點"
+                style={{ width: "100%" }}
+                classNames={{
+                    input: "border-none bg-transparent text-2xl font-semibold text-neutral-700 placeholder:text-neutral-500 focus:outline-none",
+                }}
+            />
+            <div
+                contentEditable="true"
+                className="h-full min-h-7 w-full resize-none bg-transparent text-lg font-normal text-neutral-700 placeholder:text-neutral-500 focus:outline-none"
+                ref={inputRef}
+                onInput={(e) => {
+                    Array.from(e.currentTarget.children).forEach((node) => {
+                        if (node.className.includes("pt-1.5")) return;
+                        node.classList.add("pt-1.5");
+                    });
+                    setViewpointContent(
```

Comment:
Since this trigger are quite frequent, this may result in common costly rerenders that slow the application down. Also, considering that the content of this div is not controlled, I think we can consider moving this event into onBlur